modules/afprog/afprog.c

```diff
@@ -358,18 +430,55 @@ afprogram_dd_init(LogPipe *s)
     }
   log_pipe_append(&self->super.super.super, (LogPipe *) self->writer);
 
-  return afprogram_dd_reopen(self);
+  return restore_successful ? TRUE : afprogram_dd_reopen(self);
+}
+
+static inline void
+afprogram_dd_store_reload_info(AFProgramDestDriver * const self, GlobalConfig * const cfg)
+{
```

Comment:
I have to admit, that both approach has its advantage, and it is okay for me to revert these const related changes.
@bazsi , please make a decision.